File: paintRoomCalc.py
'''
Name : paintRoomCalc in Python 3.8
Author : Аїӡек Меѥҏ
Version : v1.0
License : N/A
'''

def getLayers(roomWidth, roomDepth, roomHeight, paintThick):
    i = 0
    t = 0
    roomWidth = roomWidth * 12
    roomDepth = roomDepth * 12
    roomHeight = roomHeight * 12
    roomVol = roomWidth * roomDepth * roomHeight
    while roomVol >= 0:
        #account for walls
        wallOne = roomWidth * roomHeight
        #print("wallWidth : "+str(wallOne))
        wallThree = roomDepth * roomHeight
        #print("wallDepth : "+str(wallThree))
        wallFive = roomDepth * roomWidth
        #print("wallCeil : "+str(wallFive))

        wallVol = (wallOne * 2 + wallThree * 2 + wallFive) * paintThick
        wallArea = wallOne * 2 + wallThree * 2 + wallFive

        roomWidth = roomWidth - (paintThick * 2)
        roomDepth = roomDepth - (paintThick * 2)
        roomHeight = roomHeight - (paintThick * 2)

        #account for edges going up & down
        edgeUpVol = roomHeight*paintThick**2
        edgeUpVol = edgeUpVol*2
        edgeUpArea = roomHeight*paintThick
        edgeUpVol = edgeUpArea*2

        #account for edges going side to side {on ceiling}
        ##account for edges going side to side {depth}
        edgeSideDepthVol = (roomDepth*paintThick**2)
        edgeSideDepthVol = edgeSideDepthVol*2
        edgeSideDepthArea = (roomDepth*paintThick)
        ##account for edges going side to side {width}
        edgeSideWidthVol = roomWidth*paintThick**2
        edgeSideWidthVol = edgeSideWidthVol*2
        edgeSideWidthArea = roomWidth*paintThick
        
        #add edges
        edgeVol = edgeUpVol+edgeSideWidthVol+edgeSideDepthVol
        edgeArea = edgeUpArea+edgeSideWidthArea+edgeSideDepthArea

        #calculate final values
        roomVol = roomVol - wallVol - edgeVol
        wallArea = wallArea - edgeArea

        i += 1
        t += wallArea
    print(str(i) +
          " layers to fill your room with paint!!")
    print("& " + str(t / (4800)) + " gallons of paint")

print(Fore.LIGHTGREEN_EX + Back.BLACK + "Paint Layer Calculator")
print("by : Аїӡек Меѥҏ")
print(Style.RESET_ALL)
roomWidth = input(Fore.BLUE + "Room Width in Feet? ")
roomWidth = int(roomWidth)
roomDepth = input("Room Depth in Feet? ")
roomDepth = int(roomDepth)
roomHeight = input("Room Height in Feet? ")
roomHeight = int(roomHeight)

print("\n")
print("1 mil is 1/1000 of an inch")
paintThickMil = input("Paint Thickness in Mils? ")
paintThickMil = int(paintThickMil)
paintThick = paintThickMil / 1000
print("\n")
print("Calculating...")
getLayers(roomWidth, roomDepth, roomHeight, paintThick)

File: paintRoomCalc.fs
(*
Name : paintRoomCalc in F#
Author : Аїӡек Меѥҏ
Version : v1.0
License : N/A
*)

open System;

let getLayers (rW:float) (rD:float) (rH:float) (pT:float) : float list = 
    let paintThick = pT;
    let mutable i = 0;
    let mutable t = 0.0;
    let mutable roomWidth:float = (float rW)*12.0;
    let mutable roomDepth:float = (float rD)*12.0;
    let mutable roomHeight:float = (float rH)*12.0;
    let mutable edgeArea:float = 0.0;
    let mutable edgeSideDepthArea:float = 0.0;
    let mutable edgeSideDepthVol:float = 0.0;
    let mutable edgeSideWidthArea:float = 0.0;
    let mutable edgeSideWidthVol:float = 0.0;
    let mutable edgeUpArea:float = 0.0;
    let mutable edgeUpVol:float = 0.0; 
    let mutable edgeVol:float = 0.0;
    let mutable roomVol:float = 0.0;
    let mutable wallArea:float = 0.0;
    let mutable wallFive:float = 0.0; 
    let mutable wallOne:float = 0.0;
    let mutable wallThree:float = 0.0;
    let mutable wallVol:float = 0.0;
    let mutable roomVol:float = (float roomWidth) * (float roomDepth) * (float roomHeight);
    while (float roomVol) >= 0.0 do
        // account for walls
        wallOne <- (float roomWidth) * (float roomHeight);
        wallThree <- (float roomDepth) * (float roomHeight);
        wallFive <- (float roomDepth) * (float roomWidth);

        wallVol <- (((float wallOne) * 2.0) + ((float wallThree) * 2.0) + (float wallFive)) * (float paintThick);
        wallArea <- (((float wallOne) * 2.0) + ((float wallThree) * 2.0) + (float wallFive));

        roomWidth <- ((float roomWidth) - ((float paintThick) * 2.0));
        roomDepth <- ((float roomDepth) - ((float paintThick) * 2.0));
        roomHeight <- ((float roomHeight) - ((float paintThick) * 2.0));

        // account for edges going up & down
        edgeUpVol <- ((float roomHeight) * Math.Pow((float paintThick), 2.0));
        edgeUpVol <- ((float edgeUpVol) * 2.0);
        edgeUpArea <- ((float roomHeight) * (float paintThick));
        edgeUpVol <- ((float edgeUpArea) * 2.0);

        // account for edges going side to side {on ceiling}
        //// account for edges going side to side {depth}
        edgeSideDepthVol <- ((float roomDepth) * Math.Pow((float paintThick), 2.0));
        edgeSideDepthVol <- ((float edgeSideDepthVol) * 2.0);
        edgeSideDepthArea <- ((float roomDepth) * (float paintThick));
        //// account for edges going side to side {width}
        edgeSideWidthVol <- ((float roomWidth) * Math.Pow((float paintThick), 2.0));
        edgeSideWidthVol <- ((float edgeSideWidthVol) * 2.0);
        edgeSideWidthArea <- ((float roomWidth) * (float paintThick));

        // add edges
        edgeVol <- ((float edgeUpVol) + (float edgeSideWidthVol) + (float edgeSideDepthVol));
        edgeArea <- ((float edgeUpArea) + (float edgeSideWidthArea) + (float edgeSideDepthArea));

        // calculate final values
        roomVol <- ((float roomVol) - (float wallVol) - (float edgeVol));
        wallArea <- ((float wallArea) - (float edgeArea));

        i <- i + 1;
        t <- (float wallArea) + (float t);
    [float i; t];

    

Console.WriteLine("Paint Layer Calculator");
Console.WriteLine("by : izder456");
Console.WriteLine();

//get input for room dimentions
Console.Write("Room Width in Feet? ");
let roomWidthI:float = float(Console.ReadLine());
Console.Write("Room Depth in Feet? ");
let roomDepthI:float = float(Console.ReadLine());
Console.Write("Room Height in Feet? ");
let roomHeightI:float = float(Console.ReadLine());
Console.WriteLine();
Console.WriteLine("1 mil is 1/1000 of an inch");
Console.Write("Paint Thickness in Mils? ");
let mutable paintThickI:float = float(Console.ReadLine());
paintThickI <- float(paintThickI / 1000.0);
Console.WriteLine();
Console.WriteLine("Calculating...");

let answer = getLayers roomWidthI roomDepthI roomHeightI paintThickI;
Console.WriteLine((string answer.[0]) + " layers to fill your room with paint!!");
Console.WriteLine("& " + (string (answer.[1] / 4800.0)) + " gallons of paint");
